src/Symfony/Component/DependencyInjection/ParameterBag/EnvPlaceholderParameterBag.php

```diff
@@ -62,9 +62,17 @@ public function getEnvPlaceholders()
 
     /**
      * Merges the env placeholders of another EnvPlaceholderParameterBag.
+     *
+     * @param EnvPlaceholderParameterBag $bag
      */
     public function mergeEnvPlaceholders(self $bag)
     {
-        $this->envPlaceholders = array_merge_recursive($this->envPlaceholders, $bag->getEnvPlaceholders());
+        $newPlaceholders = $bag->getEnvPlaceholders();
+
+        foreach ($newPlaceholders as $key => $newEntries) {
```

Comment:
About the `array_map`, it works and it cuts some lines of code, but it is needless use of recursive functions and maybe not as clear, is it really needed? 
 
For the changes inside `get()`, why would you change from returning the 0th index to using foreach? If you think this should be changed how about something more readable like 

```return array_values($this->envPlaceholders[$placeholder])[0];```

Finally for the changes in setting `$this->envPlaceholders` I don't see the point in an array that will always have the structure

```
[
  'apple' => 'apple',
  'orange' => 'orange'
]
```
Plus (and I know it's going outside the scope of this PR) I don't think that assignment in a return statement is very clear and would prefer something like:

```
$uniqueName = md5($name.uniqid(mt_rand(), true));
$placeholder = sprintf('env_%s_%s', $environmentVariableName, $uniqueName);
$this->envPlaceholders[$environmentVariableName][] = $placeholder;

return $placeholder;
```

I made some naming changes locally because I also found `env`  confusing as it conflicts with Symfony environments but these don't have to be included in any PR changes.

Sorry to sound like a dick and that I don't want to change it, but I think I should at least be sure about the changes before I make them. I think it's a really cool feature and the only reason I'm here is because when  I was working on changing my own small project to use the `SYMFONY__ENV` variable style I came across this and thought it looks much better.